$NetBSD: patch-aa,v 1.4 2022/04/25 16:17:48 tnn Exp $

IRIX: gethostname takes a size_t, EAGAIN equals EWOULDBLOCK, FIOCLEX is in <sys/filio.h>, no SA_INTERRUPT.

--- socket++/local.h.orig
+++ socket++/local.h
@@ -90,8 +90,10 @@
 extern "C" char* inet_ntoa (in_addr ina);
 
 #if !defined (__linux__)
+#if !defined (__NetBSD__) && !defined (__DragonFly__) && !defined(__sun) && !defined(__sgi)
   extern "C" int gethostname (char* hostname, int len);
-#if !(defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__bsdi__) || defined(__APPLE__))
+#endif
+#if !(defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__bsdi__) || defined(__APPLE__) || defined(__DragonFly__))
   extern char* SYS_SIGLIST [];
 #endif
 #endif
